src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -775,8 +786,8 @@ private unsafe void IOLoop()
 
             while (IsOpen && !eofReceived && !_ioLoopFinished)
             {
-                bool hasPendingReads = _readQueue.Count > 0;
-                bool hasPendingWrites = _writeQueue.Count > 0;
+                bool hasPendingReads = !_readQueue.IsEmpty;
+                bool hasPendingWrites = !_writeQueue.IsEmpty;
 
                 bool hasPendingIO = hasPendingReads || hasPendingWrites;
```

Comment:
Is there a reason to store these into locals rather than just making this be:
```C#
bool hasPendingIO = !_readQueue.IsEmpty || !_writeQueue.IsEmpty;
```
?  From my perspective, that makes the code easier to read, but there's also a (small) perf benefit, in that we don't have to check whether there are any writes if there are reads.

Reply:
will create follow up PR with this change

Reply:
actually no, I use them later when calling poll

Reply:
Ok.